### PR TITLE
add extra checks for findings to dns report

### PR DESCRIPTION
**Pull request by underdarknl**
### Changes
Lets add a findings block to the dns report.

### Demo
_Please add some proof in the form of screenshots or screen recordings to show (off) new functionality, if there are interesting new features for end-users._

---

### Code Checklist
- [ ] All the commits in this PR are properly PGP-signed and verified.
- [ ] This PR only contains functionality relevant to the issue; tickets have been created for newly discovered issues.
- [ ] I have written unit tests for the changes or fixes I made.
- [ ] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [ ] I have performed a self-review of my code and refactored it to the best of my abilities.

### Communication
- [ ] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [ ] I have made corresponding changes to the documentation, if necessary.
- [ ] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---
## Checklist for code reviewers:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---
## Checklist for QA:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

**Review by ammar92 — DISMISSED** [review]

File: rocky/reports/report_types/dns_report/report.py
Comment (dekkers) [inline rocky/reports/report_types/dns_report/report.py:30]:
```suggestion
        finding_types: dict[str, dict] = {}
```

File: rocky/reports/report_types/dns_report/report.py
Comment (dekkers) [inline rocky/reports/report_types/dns_report/report.py:69]:
```suggestion
        finding_types_sorted = sorted(finding_types.values(), key=lambda x: x["finding_type"].risk_score or 0, reverse=True)
```

File: rocky/reports/report_types/dns_report/report.py
Comment (dekkers) [inline rocky/reports/report_types/dns_report/report.py:73]:
```suggestion
        return {"input_ooi": input_ooi, "records": records, "security": security, "finding_types": finding_types_sorted}
```

**Review by dekkers — COMMENTED** [review]
Looks good to me. My suggestions should fix the mypy errors, but somebody else should commit them because if I commit them I can't approve the PR because I'm the last committer and Jan can't because he created the PR, so that doesn't work...

**Comment by dekkers** [comment]
## Checklist for QA:
- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [ x I confirmed that there are no unintended functional regressions in this branch:
    - [ ] I have managed to pass the onboarding flow
    - [x] Objects and Findings are created properly
    - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.

### What works:
* DNS report works and shows the findings
image

**Review by dekkers — APPROVED** [review]

**Pull request merged**
